### DISCUSSION TITLE
On kernel_size_factor for training an SSL model for streaming ASR

### DISCUSSION THREAD

**Discussion by piraka9011** [NVIDIA-NeMo/NeMo#4579] [General]
I'd like to pre-train a self supervised model with my dataset (90k hours) which could then be used to fine tune a CitriNet large model.
I'd like to use this model in an ASR streaming task.

Does it make sense for me to set the `kernel_size_factor` in the `jasper` blocks of the encoder to `0.25` for the `SpeechEncDecSelfSupervisedModel` or should I do that when fine tuning the `EncDecCTCModelBPE`? (I'm assuming it's the former but wanted to x2 check.)

I also noticed the default config for the optimizer having a pretty high LR:

```yaml
lr: 5.0
# optimizer arguments
betas: [0.9, 0.98]
weight_decay: 1e-3
```

as well as the scheduler using `NoamAnnealing` instead of `CosineAnnealing`.
Is this the recommended config?

**Comment by titu1994** [comment]
Kernel size factor affects the weights of the trained model, you cannot change it after you finish training. So yes, you would need to set it for both SSL and fine tuning. 

For streaming, if might even be better to use Conformer large, it usually converges faster and is generally quite easily streamable using Riva. But it can't handle long files. 

SSL training requires different training config than finetuning of the underlying model, id generally stick to the SSL config defaults

**Reply by piraka9011** [reply]
Thank you for the quick reply!

re: Conformer large, I noticed that it performs better than CitriNet, but Riva does not support Conformer-RNNT, only [Conformer-CTC](https://docs.nvidia.com/deeplearning/riva/user-guide/docs/reference/models/asr.html#conformer-ctc).

If you're suggesting to use the CTC version, then I can definitely try [the suggested configuration](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/ssl/conformer/conformer_ssl.yaml).

As long as it can stream long sessions, that is fine. We don't need to support transcription of long files/offline transcription at this time.

**Reply by titu1994** [reply]
You can use Conformer CTC training as well, it attains fast convergence and can be streamed with riva. We found that it's streaming capability is usually better than even Citrinet.
